libp2p/stream/bufferstream.nim

```diff
@@ -108,13 +112,14 @@ method pushEof*(s: BufferStream) {.base, async.} =
 method atEof*(s: BufferStream): bool =
   s.isEof and s.readBuf.len == 0
 
-method readOnce*(s: BufferStream,
-                 pbytes: pointer,
-                 nbytes: int):
-                 Future[int] {.async.} =
+method readOnce*(
+    s: BufferStream,
+    pbytes: pointer,
+    nbytes: int
+): Future[int] {.async: (raises: [CancelledError, LPStreamError]).} =
   doAssert(nbytes > 0, "nbytes must be positive integer")
   doAssert(not s.reading,
```

Comment:
should we be raising Defects here?

Reply:
Defects can always be raised, regardless of `{.async: (raises).}` annotation. they are not newly introduced